lecture/src/main/java/de/tum/in/www1/artemis/lecture/web/rest/ExerciseUnitResource.java

```diff
@@ -0,0 +1,117 @@
+
+    private final AuthorizationCheckService authorizationCheckService;
+
+    private final ExerciseUnitRepository exerciseUnitRepository;
+
+    private final LectureRepository lectureRepository;
+
+    @Value("${jhipster.clientApp.name}")
+    private String applicationName;
+
+    public ExerciseUnitResource(LectureRepository lectureRepository, ExerciseUnitRepository exerciseUnitRepository, AuthorizationCheckService authorizationCheckService) {
+        this.exerciseUnitRepository = exerciseUnitRepository;
+        this.lectureRepository = lectureRepository;
+        this.authorizationCheckService = authorizationCheckService;
+    }
+
+    /**
+     * POST lectures/:lectureId/exercise-units : creates a new exercise unit.
+     *
+     * @param lectureId    the id of the lecture to which the attachment unit should be added
+     * @param exerciseUnit the exercise unit that should be created
+     * @return the ResponseEntity with status 201 (Created) and with body the new exercise unit
+     * @throws URISyntaxException if the Location URI syntax is incorrect
+     */
+    @PostMapping("lectures/{lectureId}/exercise-units")
+    @PreAuthorize("hasRole('EDITOR')")
+    public ResponseEntity<ExerciseUnit> createExerciseUnit(@PathVariable Long lectureId, @RequestBody ExerciseUnit exerciseUnit) throws URISyntaxException {
+        log.debug("REST request to create ExerciseUnit : {}", exerciseUnit);
+        if (exerciseUnit.getId() != null) {
+            throw new BadRequestAlertException("An exercise unit must have an ID to be updated", ENTITY_NAME, "idnull");
+        }
+        Optional<Lecture> lectureOptional = lectureRepository.findByIdWithPostsAndLectureUnitsAndLearningGoals(lectureId);
+        if (lectureOptional.isEmpty()) {
+            throw new BadRequestAlertException("Invalid id", ENTITY_NAME, "not found");
+        }
+        Lecture lecture = lectureOptional.get();
+        if (lecture.getCourse() == null) {
+            throw new ConflictException();
+        }
+        if (!authorizationCheckService.isAtLeastEditorInCourse(lecture.getCourse(), null)) {
+            throw new AccessForbiddenException(NOT_ALLOWED);
+        }
+
+        // persist lecture unit before lecture to prevent "null index column for collection" error
+        exerciseUnit.setLecture(null);
+        exerciseUnit = exerciseUnitRepository.saveAndFlush(exerciseUnit);
+        exerciseUnit.setLecture(lecture);
+        lecture.addLectureUnit(exerciseUnit);
+        Lecture updatedLecture = lectureRepository.save(lecture);
+        ExerciseUnit persistedExerciseUnit = (ExerciseUnit) updatedLecture.getLectureUnits().get(updatedLecture.getLectureUnits().size() - 1);
+
+        return ResponseEntity.created(new URI("/api/exercise-units/" + persistedExerciseUnit.getId()))
+                .headers(HeaderUtil.createEntityCreationAlert(applicationName, true, ENTITY_NAME, "")).body(persistedExerciseUnit);
```

Comment:
Same translation issue as commented on the video-units applies here as well.

Reply:
Addressed in [#4466](https://github.com/ls1intum/Artemis/pull/4466/commits/568ef4cc8b60e5ab1b64819faf5829cfecea32dd)